testing/async-errors.js

```diff
@@ -29,6 +30,11 @@ function stubAsyncErrorThrows() {
   rethrowAsyncSandbox = sinon.createSandbox();
   rethrowAsyncSandbox.stub(coreError, 'rethrowAsync').callsFake((...args) => {
     const error = coreError.createErrorVargs.apply(null, args);
+    const index = indexOfExpectedMessage(error.message);
+    if (index != -1) {
+      expectedAsyncErrors.splice(index, 1);
+      return;
```

Comment:
Clever! This will prevent errors from leaking from tests, and could really improve reliability.

/cc @rileyajones as an FYI.